lib/customisation.nix

```diff
@@ -200,7 +200,9 @@ rec {
       # Only show the error for the first missing argument
       error = errorForArg missingArgs.${head (attrNames missingArgs)};
 
-    in if missingArgs == {} then makeOverridable f allArgs else abort error;
+    in if missingArgs == {}
+       then makeOverridable f allArgs
+       else throw "lib.customisation.callPackageWith: ${error}";
```

Comment:
What was the motivation for this specific change? It makes it so `nix-env -qa` no longer fails and just silently drops the attrpath when querying pacakges that have missing arguments (and therefore ofborg and other CI relies on the previous behavior to catch missing argument errors during outpath calculation)

This still smells like a bug in `nix-env` regardless to me, so this bit in Nixpkgs probably doesn't need to be changed back, but I'm still trying to trace down in that code where/why this throw would be ignored rather than propagated as an eval error

Reply:
Nope, this is intended behavior in Nix:

> Throw an error message *s*. This usually aborts Nix expression
> evaluation, but in `nix-env -qa` and other commands that try to
> evaluate a set of derivations to get information about those
> derivations, a derivation that throws an error is silently skipped
> (which is not the case for `abort`).

https://nixos.org/manual/nix/stable/language/builtins.html#builtins-throw

Reply:
Oh damn yeah. Probably the entire reason this is in nix-env at all is because we historically haven't had anything like this PR, which would've prevented packages from not being evaluatable.

Reply:
Revert and eval fixes in https://github.com/NixOS/nixpkgs/pull/278777